memory_statistics/memory_statistics.py

```diff
@@ -167,16 +163,28 @@ def validate_library_config(config):
     if "compiler_flags" not in config:
         config["compiler_flags"] = []
 
+def parse_src_input_to_file_name_map(src):
+    ret = {}
+    for s in src:
+        if isinstance(s, str):
+            ret[s] = os.path.basename(s)
+        else:
+            ret[s['file']] = f"{os.path.basename(s['file'])} ({s['tag']})"
```

Comment:
Is this logic concatenating the file name and its tag in a string? A comment will be helpful